plentyofpets/lib/components/user_profile.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+
+class UserProfile extends StatelessWidget {
+  const UserProfile({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    CollectionReference users = FirebaseFirestore.instance.collection('users');
+
+    return Container(
+        margin: const EdgeInsets.all(10.0),
+        child: FutureBuilder<DocumentSnapshot>(
+          future: users.doc('GP11dBKUW7C1Xj5OeyHV').get(),
+          builder:
+              (BuildContext context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+            if (snapshot.hasError) {
+              return Text("Something went wrong: ${snapshot.error}");
+            }
+
+            if (snapshot.hasData && !snapshot.data!.exists) {
+              return const Text("Document does not exist");
+            }
+
+            if (snapshot.connectionState == ConnectionState.done) {
+              Map<String, dynamic> data =
+                  snapshot.data!.data() as Map<String, dynamic>;
+
+              return Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text('Name: ${data['firstName']}' +
+                        ' ' +
+                        '${data['lastName']}'),
```

Comment:
Do we need to use the template string here?  I kind of expected last name to be a string already. Same with the state on line 37

Reply:
You're totally right, I've refactored and removed the template string for last name and state.